layouts/partials/header.html

```diff
@@ -1,8 +1,10 @@
 <nav class="navigation">
   <section class="container">
-    <a class="navigation-title" href="{{ .Site.BaseURL | relLangURL }}">
-      {{ .Site.Title }}
+    {{ with .Site }}
+    <a class="navigation-title" href="{{ index (apply (slice .Home.RelPermalink) (.Params.urlFunc | default "relLangURL") ".") 0 }}">
```

Comment:
I think this could be simpler.  https://gohugo.io/functions/urls/abslangurl/
It seems `{{ absLangURL "" }}` fixes the issue. Could you try that?